### PR TITLE
ci(cypress): Use cypress-split for running tests in parallel

### PR DESCRIPTION
**Pull request by juliushaertl**
Not as equally distributed as with cypress cloud but much better than waiting for over an hour on one job to finish.

current distribution
Screenshot 2023-10-11 at 20 33 52

### CODE REVIEW

**Comment by juliushaertl** [comment]
Job summary is not optimal likely due to our custom specPattern config (ref https://github.com/bahmutov/cypress-split/issues/69)

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -83,23 +83,6 @@ jobs:
         containers: [1, 2, 3, 4, 5, 6, 7, 8]
         php-versions: [ '8.1' ]
         server-versions: [ 'master' ]
-        run-in-parallel:
```
Comment (mejo-) [inline .github/workflows/cypress.yml:86]: `run-in-parallel` is used further down in the workflow, so I think it's worth keeping it for now. Maybe we can change this to `${{ !!github.head_ref }}` again, to make sure that CI runs on development+release branches don't register in the Cypress cloud - and only the PR CI jobs end up in Cypress cloud.

**Review by mejo- — CHANGES_REQUESTED** [review]

File: cypress.config.js [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+const cypressSplit = require('cypress-split')
+
+module.exports = defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      // IMPORTANT: return the config object
+      return config
+    },
+  },
+})
```
Comment (skjnldsv) [inline cypress.config.js:11]:
```suggestion
const cypressSplit = require('cypress-split')

module.exports = defineConfig({
	e2e: {
		setupNodeEvents(on, config) {
			// IMPORTANT: return the config object
			return config
		},
	},
})
```

Comment (skjnldsv) [inline cypress.config.js:11]:
```suggestion
const cypressSplit = require('cypress-split')

module.exports = defineConfig({
	e2e: {
		setupNodeEvents(on, config) {
			// IMPORTANT: return the config object
			return config
		},
	},
})
```

Comment (skjnldsv) [inline cypress.config.js:11]:
```suggestion
const cypressSplit = require('cypress-split')

module.exports = defineConfig({
	e2e: {
		setupNodeEvents(on, config) {
			// IMPORTANT: return the config object
			return config
		},
	},
})
```

**Comment by juliushaertl** [comment]
Good to go from my side.

The author has also already [fixed the github summary for our case](https://github.com/bahmutov/cypress-split/pull/118) already which now nicely displays spec stats:

https://github.com/bahmutov/cypress-split/actions/runs/6496437037#summary-17643354799

One more report remaining about the time but that doesn't seem too critical to me.  https://github.com/bahmutov/cypress-split/issues/121

**Review by max-nextcloud — APPROVED** [review]

**Pull request merged**